### PR TITLE
Feature/hide viewed

### LINKED ISSUE #38: [FEATURE] Add the ability to hide repositories you reviewed
### Is this a unique feature?

- [X] I have checked "open" AND "closed" issues and this is not a duplicate

### Is your feature request related to a problem/unavailable functionality? Please describe.

When looking at the list of trending repositories , the user might have to repeatedly continue scroll over repositories he already reviewed.

### Proposed Solution

Add a button to hide repositories, so they won't be shown again, possibly with a page that shows the hidden repositories (in case you accidently pressed the button)

### Do you want to work on this issue?

No

### PR DESCRIPTION
**Pull request by kvtemadden**
## Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Used `localStorage` to handle "viewed" repos in place of using a database. When a repo is clicked on, it is stored in `localStorage` in its entirety (due to API not having a `getRepoById` endpoint).

Also added a button in the top toolbar, "Hide Viewed". When clicked, viewed repos are removed from the DOM and animated out using `framer motion`. 

When hidden, a link to "viewed repos" becomes available. Repos from `localStorage` are displayed on this page.

Fixes #38 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

## Screenshots (if applicable):

Screen Recording 2024-10-03 at 15 50 11

### CODE REVIEW

**Review by Bashamega — CHANGES_REQUESTED** [review]
Please fix the build
image

**Comment by kvtemadden** [comment]
@Bashamega, apologies, fixed that oversight. Also solved merge conflicts with main.

**Comment by Bashamega** [comment]
> @Bashamega, apologies, fixed that oversight. Also solved merge conflicts with main.

No problem

File: src/components/Repos.tsx [hunk 1]
```diff
@@ -67,46 +73,71 @@ const Repos: React.FC = () => {
...
-            </SelectItem>
-            <SelectItem value="weekly">Weekly</SelectItem>
-            <SelectItem value="monthly">Monthly</SelectItem>
-          </SelectContent>
-        </Select>
+          {showViewed ? "Hide Viewed" : "Show Viewed"}
+        </button>
       </div>
 
+      <AnimatePresence>
```
Comment (Bashamega) [inline src/components/Repos.tsx:120]: Make it in either way viewed

File: src/app/repos/viewed/page.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      const storedRepos = localStorage.getItem("ttViewedRepos");
+      const parsedData = storedRepos ? JSON.parse(storedRepos) : [];
+      setData(parsedData);
+    }
+  }, []);
+
+  return (
+    <main>
+      <Navbar />
+      {data ? (
```
Comment (Bashamega) [inline src/app/repos/viewed/page.tsx:23]: Add search filtration here also

File: src/app/repos/viewed/page.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        data.length > 0 ? (
+          <div className="grid gap-5 px-5 pt-5">
+            {data.map((item: GithubData, index: number) => (
+              <Repo data={item} key={index} show={true} />
```
Comment (Bashamega) [inline src/app/repos/viewed/page.tsx:27]: Add the date of when it was viewed

File: src/components/Repo.tsx [hunk 4]
```diff
@@ -9,47 +11,87 @@ import { getBestContrastColor } from "@/lib/contrastColors";
...
+      setViewed(false);
+    }
+  }, [data]);
+
+  const handleOnClick = () => {
+    const existingRepos = localStorage.getItem("ttViewedRepos");
+    const repos = existingRepos ? JSON.parse(existingRepos) : [];
+
+    if (!repos.includes(data.id)) {
+      repos.push(data);
```
Comment (Bashamega) [inline src/components/Repo.tsx:40]: also add the current date, so you can show when you have added the repo

**Review by Bashamega — CHANGES_REQUESTED** [review]
Great work just some little tweaks

File: src/components/Repos.tsx (hunk 1 above)
Comment (kvtemadden) [reply src/components/Repos.tsx:120]: Hey - just fixed the other comments, can you clarify what you mean on this one please?

File: src/app/repos/viewed/page.tsx (hunk 2 above)
Comment (kvtemadden) [reply src/app/repos/viewed/page.tsx]: ✅ - also added in date filter to filter based on date.

File: src/app/repos/viewed/page.tsx (hunk 3 above)
Comment (kvtemadden) [reply src/app/repos/viewed/page.tsx:71]: ✅

File: src/components/Repo.tsx (hunk 4 above)
Comment (kvtemadden) [reply src/components/Repo.tsx]: ✅

File: src/components/Repos.tsx (hunk 1 above)
Comment (Bashamega) [reply src/components/Repos.tsx:120]: So currently, the button only shows when you click on the other button. I want the 2 buttons to be visible

Comment (kvtemadden) [reply src/components/Repos.tsx]: ✅ Gotcha, updated that.

**Comment by Bashamega** [comment]
Please resolve the conflict

**Review by Bashamega — APPROVED** [review]

**Comment by Bashamega** [comment]
Great work!

**Pull request merged**